interfaces/machine_parameters/ring_section.py

```diff
@@ -177,98 +165,134 @@ def __init__(self, section_length, alpha_0,
 
         # Reshaping the input synchronous data to the adequate format and
         # get back the momentum program
-        if not isinstance(synchronous_data, dTypes._ring_program):
+        if not isinstance(synchronous_data,
+                          ring_programs._synchronous_data_program):
             synchronous_data \
-                = dTypes._ring_program.conversions[func_type](synchronous_data)
+                = ring_programs._synchronous_data_program._conversions[
+                    func_type](synchronous_data)
 
         self.synchronous_data = synchronous_data
 
+        # Setting orbit length
+        if orbit_bump is None:
+            self.length = ring_programs.orbit_length_program(
+                self.length_design)
+        else:
+            if not isinstance(orbit_bump, ring_programs.orbit_length_program):
+                orbit_bump = ring_programs.orbit_length_program(orbit_bump)
+            if orbit_bump.timebase == 'by_time':
+                orbit_bump[:, 1, :] += self.length_design
+            else:
+                orbit_bump += self.length_design
+            self._check_and_set_alpha_and_orbit(orbit_bump)
+
         # Setting the linear momentum compaction factor
         # Checking that the synchronous data and the momentum compaction
         # have the same length if defined turn-by-turn, raise a warning if one
         # is defined by turn and the other time based
-        self.alpha_order = 0
-        if not isinstance(alpha_0, dTypes.momentum_compaction):
-            alpha_0 = dTypes.momentum_compaction(alpha_0, order=0)
+        self.alpha_orders = [0]
+        alpha_order_max = 0
+        if not isinstance(alpha_0, ring_programs.momentum_compaction):
+            alpha_0 = ring_programs.momentum_compaction(alpha_0, order=0)
         else:
             if alpha_0.order != 0:
                 raise excpt.InputError(
                     "The order of the datatype passed as keyword " +
                     "argument alpha_%s do not match" % (0))
 
-        self._check_and_set_momentum_compaction(alpha_0, 0)
+        self._check_and_set_alpha_and_orbit(alpha_0, 0)
 
         # Treating non-linear momentum compaction factor if declared
         # Listing all the declared alpha first
         alpha_n = {1: alpha_1, 2: alpha_2}
```

Comment:
You can change this to
```suggestion
        alpha_n = {1: alpha_1, 2: alpha_2, **{k:kwargs[k] for k in kwargs if 'alpha_' in k}}
```
And then do all the alphas in one go.

Reply:
Impressive! I think I'll keep the \_\_init\_\_ a bit more verbose for now in case a new dev in the future tries to read through, it will be easier, but for the `direct_input` method since this is in line with the suggestions few comments above I'll give it a try

Reply:
You do over long names, I do over dense code!